napari_tiff/napari_tiff_metadata.py

```diff
@@ -0,0 +1,395 @@
+from typing import Any
+
+import numpy
+from tifffile import PHOTOMETRIC, TiffFile, xml2dict
+from vispy.color import Colormap
+
+from napari_tiff.napari_tiff_colormaps import (
+    alpha_colormap,
+    cmyk_colormaps,
+    int_to_rgba,
+    rgb_colormaps,
+)
+
+
+def get_metadata(tif: TiffFile) -> dict[str, Any]:
+    """Return metadata keyword argument dictionary for napari layer."""
+    if tif.is_ome:
+        metadata_kwargs = get_ome_tiff_metadata(tif)
+    # TODO: combine interpretation of imagej and tags metadata?:
+    elif tif.is_imagej:
+        metadata_kwargs = get_imagej_metadata(tif)
+    else:
+        metadata_kwargs = get_tiff_metadata(tif)
+
+    # napari does not use this this extra `metadata` layer attribute
+    # but storing the information on the layer next to the data
+    # will allow users to access it and use it themselves if they wish
+    metadata_kwargs["metadata"] = get_extra_metadata(tif)
+
+    return metadata_kwargs
+
+
+def get_extra_metadata(tif: TiffFile) -> dict[str, Any]:
+    """Return any non-empty tif metadata properties as a dictionary."""
+    metadata_dict = {}
+    empty_metadata_values = [None, "", (), []]
+    for name in dir(tif.__class__):
+        obj = getattr(tif.__class__, name)
+        if "metadata" in name:
```

Comment:
```suggestion
        if "metadata" in name:
            obj = getattr(tif.__class__, name)
```

Reply:
Nice catch, thanks.

Probably it's even better if I make it a single line:
```
metadata_value = getattr(tif.__class__, name).__get__(tif)
```
rather than:
```
obj = getattr(tif.__class__, name)
metadata_value = obj.__get__(tif)
```

Reply:
Done.